recipes/faiss/all/patches/fix-cmake-1.7.4.patch

```diff
@@ -0,0 +1,252 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index fa1e3124..2f1b07a8 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -18,10 +18,17 @@ set(CMAKE_CXX_STANDARD 11)
+ list(APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
+ 
+ # Valid values are "generic", "avx2".
+-option(FAISS_OPT_LEVEL "" "generic")
++option(FAISS_WITH_OPT_GENERIC "Building a library without optimization" OFF)
++option(FAISS_WITH_OPT_AVX2 "Building a library with avx2 optimization" ON)
+ option(FAISS_ENABLE_GPU "Enable support for GPU indexes." ON)
+ option(FAISS_ENABLE_PYTHON "Build Python extension." ON)
+ option(FAISS_ENABLE_C_API "Build C API." OFF)
++option(BUILD_SHARED_LIBS "Building shared library" ON)
++option(FAISS_WITH_PIC "Building static with pic" ON)
++
++if(BUILD_SHARED_LIBS)
++  set(FAISS_WITH_PIC ON)
++endif()
+ 
+ if(FAISS_ENABLE_GPU)
+   set(CMAKE_CUDA_HOST_COMPILER ${CMAKE_CXX_COMPILER})
+diff --git a/faiss/CMakeLists.txt b/faiss/CMakeLists.txt
+index 1fea676c..120107f5 100644
+--- a/faiss/CMakeLists.txt
++++ b/faiss/CMakeLists.txt
+@@ -221,83 +221,75 @@ endif()
+ # Export FAISS_HEADERS variable to parent scope.
+ set(FAISS_HEADERS ${FAISS_HEADERS} PARENT_SCOPE)
+ 
+-add_library(faiss ${FAISS_SRC})
+-
+-add_library(faiss_avx2 ${FAISS_SRC})
+-if(NOT FAISS_OPT_LEVEL STREQUAL "avx2")
+-  set_target_properties(faiss_avx2 PROPERTIES EXCLUDE_FROM_ALL TRUE)
+-endif()
+-if(NOT WIN32)
+-  target_compile_options(faiss_avx2 PRIVATE $<$<COMPILE_LANGUAGE:CXX>:-mavx2 -mfma -mf16c -mpopcnt>)
+-else()
+-  # MSVC enables FMA with /arch:AVX2; no separate flags for F16C, POPCNT
+-  # Ref. FMA (under /arch:AVX2): https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64
+-  # Ref. F16C (2nd paragraph): https://walbourn.github.io/directxmath-avx2/
+-  # Ref. POPCNT: https://docs.microsoft.com/en-us/cpp/intrinsics/popcnt16-popcnt-popcnt64
+-  target_compile_options(faiss_avx2 PRIVATE $<$<COMPILE_LANGUAGE:CXX>:/arch:AVX2>)
+-  # we need bigobj for the swig wrapper
+-  add_compile_options(/bigobj)
++find_package(OpenMP REQUIRED)
```

Comment:
This will fail on AppleClang and on Clang provided by most Linux package managers.

Related:
- #24584
- #22360